src/SOS/extensions/hostcoreclr.cpp

```diff
@@ -709,22 +701,40 @@ bool SetHostRuntimeDirectory(LPCSTR hostRuntimeDirectory)
             return false;
         }
         hostRuntimeDirectory = _strdup(fullPath.c_str());
+
+        // Try to get the runtime version from the host runtime directory
+        if (major == 0)
+        {
+            uint32_t majorVersion = 0;
+            uint32_t minorVersion = 0;
+            uint32_t revision = 0;
+            if (sscanf(GetFileName(fullPath).c_str(), "%d.%d.%d", &majorVersion, &minorVersion, &revision) == 3)
```

Comment:
Does this work well with semver preview versions?

Reply:
Yes.

Reply:
If they still have 9.0.0-preview.....

Reply:
We use the same code in FindDotNetVersion which I think you wrote.

Reply:
That code probes well known locations. This one trusts user inputs and changes loading behavior on it. Was trying to think if this can be attack surface, but can't think a case where this adds issues that didn't already exist.